lib/unleash_proxy_client_flutter.dart

```diff
@@ -257,7 +257,27 @@ class UnleashClient extends EventEmitter {
     }
   }
 
+  /// Checks if any of the provided context fields are different from the current ones.
+  bool _anyFieldHasChanged(Map<String, String> fields) {
+    for (var entry in fields.entries) {
+      String key = entry.key;
+      String newValue = entry.value;
+
+      if (key == 'userId') {
+        if (context.userId != newValue) return true;
+      } else if (key == 'sessionId') {
+        if (context.sessionId != newValue) return true;
+      } else if (key == 'remoteAddress') {
+        if (context.remoteAddress != newValue) return true;
+      } else {
+        if (context.properties[key] != newValue) return true;
```

Comment:
I'm guessing this one also handles cases where the key didn't exist before? That is, if `context.properties[key]` is null/undefined/whatever it is in dart, then null != "some string" kicks in and we'll run the update, right? 

I don't see a unit test for that, but it might be worth adding one (or modifying the current one)?

Reply:
Very good point. I will add a test and I think we're missing field removal in update context